### PR TITLE
[4.0] [a11y] proof of concept for admin tables

### PR DESCRIPTION
**Pull request by brianteeman**
This is a proof of concept so that the a11y team can check it and so that hopefully a better developer can improve the js and php
ONLY com_content?view=articles for testing

Part 1. 
Silly popover that says you can click to sort is replaced by the standard arrows
aria-sorted added to show current sort status ascending|descending|none to assisted technology

Part 2.
All tables should have a caption
A caption is added that describes the table and the current sort
eg Table of articles , sorted by author descending
This has an aria-live attribute which we don't actually need as the page is always reloaded on sort but if we ever switch to an ajax sort it will be needed so that whenever the sorting is changed the caption is updated and announced to screenreaders

Part 3.
The sort element has been changed from an <a to a semantically correct button
 
#### Testing 
If not using npm -i this can still be tested but the sort buttons will not be styled

#### TODO
Move the scripts to searchtools.js 
_**Convert one script from jquery to es6 - help appreciated**_
Add caption (and language string) for each table

#### B/C
The majority of the changes are in an updated searchtools layout and (eventually) an updated searchtools.js
So the only change a developer needs to do to take advantage of this increased accessibility is to add the caption to the list and an appropriate language string

### CODE REVIEW

File: layouts/joomla/searchtools/grid/sort.php [hunk 1]
```diff
@@ -13,20 +13,35 @@
 use Joomla\CMS\Language\Text;
 
 $data = $displayData;
+$icon = "icon-menu-2";
+$sort = "none";
+$caption = '';
+$selected ='';
```
Comment (Quy) [inline layouts/joomla/searchtools/grid/sort.php]: Add space after `=` and next line.

File: layouts/joomla/searchtools/grid/sort.php [hunk 2]
```diff
@@ -13,20 +13,35 @@
...
+$id ='';
 
-$title = htmlspecialchars(Text::_($data->tip ?: $data->title));
-HTMLHelper::_('bootstrap.popover');
+if ($data->order === $data->selected) :
+    $icon = $data->orderIcon;
+    $sort = ($data->direction === "asc" ? "ascending" : "descending");
+    $caption = !empty($data->title) ? Text::_($data->title) . ' - ' . $sort : '';
+    $selected = " selected";
+    $id = "id=\"sorted\"";
```
Comment (Quy) [inline layouts/joomla/searchtools/grid/sort.php]: Change outside double quotes to single quotes then you don't need to escape the inside double quotes. Change the same for lines 24 & 26.

File: layouts/joomla/searchtools/grid/sort.php [hunk 3]
```diff
@@ -13,20 +13,35 @@
...
+$sort = "none";
+$caption = '';
+$selected ='';
+$id ='';
 
-$title = htmlspecialchars(Text::_($data->tip ?: $data->title));
-HTMLHelper::_('bootstrap.popover');
+if ($data->order === $data->selected) :
+    $icon = $data->orderIcon;
+    $sort = ($data->direction === "asc" ? "ascending" : "descending");
```
Comment (Quy) [inline layouts/joomla/searchtools/grid/sort.php]: Remove `()` as they are not necessary.

**Comment by brianteeman** [comment]
Working on converting the columnheader sort to be a real button -

**Comment by brianteeman** [comment]
Forgot to update the comment that it is now correctly a button

**Comment by brianteeman** [comment]
Forgot to add that this is inline with the new proposed template design by @coolcat-creations which calls for the addition of sort arrows

File: layouts/joomla/searchtools/grid/sort.php (hunk 2 above)
Comment (brianteeman) [reply layouts/joomla/searchtools/grid/sort.php]: done

**Comment by brianteeman** [comment]
Conflicts fixed

**Comment by brianteeman** [comment]
Something broke - the aria-sorted atribute is not being added correctly. Working on that today

**Comment by brianteeman** [comment]
Closed - obviously no one else is interested in improving the accessibility in joomla 4

**Pull request closed without merging**